kngsild/src/test/kotlin/io/egm/kngsild/utils/AuthUtilsTest.kt

```diff
@@ -44,19 +40,36 @@ class AuthUtilsTest {
     }
 
     @Test
-    fun `it should return an access token`() {
+    fun `it should return an access token with ClientCredentials`() {
         stubFor(
             post(urlMatching("/auth"))
                 .willReturn(
                     ok().withBody(serializeObject(mapOf("access_token" to "token")))
                 )
         )
-
-        val response = AuthUtils(
+        val clientCredentials = ClientCredentials(
             "http://localhost:8089/auth",
             "client_id",
             "client_secret",
             "client_credentials"
+        )
+
+        val response = AuthUtils(
+            clientCredentials = clientCredentials,
+            authType = AuthType.CLIENT_CREDENTIALS
+        ).getToken()
+
+        assertTrue(response.isRight())
+        assertEquals(response, "token".right())
+    }
+
+    @Test
+    fun `it should return an access token with ProvidedToken`() {
```

Comment:
it does not actually **return** a thing, it throws an exception, this is different